File: src/api/operations/localStorage.ts
import { invoke } from "@tauri-apps/api";

class LocalStorage {
  async getItem<T>(key: string): Promise<T | null> {
    return await invoke<T | null>("get_item", { key });
  }
  async setItem<T>(key: string, value: T) {
    return await invoke("set_item", { key, value });
  }
}

const storage = new LocalStorage();

export default storage;

File: src-tauri/src/commands/local_storage.rs
use crate::storage::LocalStorage;
use serde_json::Value;
use tauri::{command, AppHandle};
use tracing::instrument;

#[command]
#[instrument(name = "GetStorage", skip(app))]
pub fn get_item(key: String, app: AppHandle) -> Value {
    LocalStorage::get_this(app).get(&key)
}

#[command]
#[instrument(name = "SetStorage", skip(app), err)]
pub fn set_item(key: String, value: Value, app: AppHandle) -> Result<(), String> {
    LocalStorage::get_this(app).set(&key, value)
}
